src/weight_windows.cpp

```diff
@@ -815,13 +787,59 @@ WeightWindowsGenerator::WeightWindowsGenerator(pugi::xml_node node)
   // create a matching weight windows object
   auto wws = WeightWindows::create();
   ww_idx_ = wws->index();
+  wws->set_mesh(mesh_idx);
   if (e_bounds.size() > 0)
     wws->set_energy_bounds(e_bounds);
-  wws->set_mesh(model::mesh_map[mesh_id]);
   wws->set_particle_type(particle_type);
   wws->set_defaults();
 }
 
+void WeightWindowsGenerator::create_tally()
+{
+  const auto& wws = variance_reduction::weight_windows[ww_idx_];
+
+  // create a tally based on the WWG information
+  Tally* ww_tally = Tally::create();
+  tally_idx_ = model::tally_map[ww_tally->id()];
+  ww_tally->set_scores({"flux"});
```

Comment:
for future methods, we may need to refactor this if we are thinking about using other scores for generating weight windows. Perhaps the `create_tally` method takes the method and it's parameters as the argument.

Reply:
For sure. I went back and forth on whether or not to allow scores here, but with the original changeset already being pretty large I decided that was a task that could be tackled separately.

Reply:
Created an issue to track this: #2584. I think it requires a little more discussion and identification of use cases, but I'm notionally on board.